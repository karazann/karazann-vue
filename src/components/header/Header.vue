<template lang="pug">
    mixin menu-item(linkTo, text)
        li
            n-link(class!=attributes.class, to=linkTo)=text

    header(:class="{'shadow': !top || shadow}")
        .container
            .row.inner
                v-brand
                .links
                    div(v-if='!signedIn')
                        .horizontal-list
                            +menu-item('/developers', 'Developers')
                            +menu-item('/blog', 'Blog')
                            +menu-item('/faq', 'FAQ')
                            +menu-item('/roadmap', 'Roadmap')
                            +menu-item('/auth/signin', 'Sign in')
                            +menu-item('/auth/signup', 'Sign up')(class='btn')
                    div(v-if='signedIn')
                        .horizontal-list
                            +menu-item('/', 'Hello')
                            +menu-item('/', 'Test')
                        .horizontal-list
                            li
                                profile-menu(:user='user')
</template>

<script>
    import ProfileMenu from './ProfileMenu.vue'

    export default {
        name: 'v-header',
        components: {
            ProfileMenu
        },
        props: {
            top: Boolean,
            signedIn: Boolean,
            shadow: Boolean
        }
    }
</script>

<style lang="scss" scoped>
    header {
        top: 0;
        display: flex;
        padding: 20px 0;
        justify-content: space-between;
        background: #fff;
        position: relative;
        position: fixed;
        width: 100%;
        z-index: 1000;
        transition: all 0.4s ease;
    }
    .shadow {
        box-shadow: 0 3px 30px rgba(0, 0, 0, 0.08);
    }
    .inner {
        display: flex;
        justify-content: space-between;
        align-items: center;
    }
    .horizontal-list {
        display: flex;
        list-style: none;
    }
    .links {
        display: none;
        position: relative;

        @include media-breakpoint-up(md) {
            // Larger than md: 768px,
            display: flex;
        }

        a {
            margin: 0 15px 0 22px;
            height: 38px;
            line-height: 38px;
            text-decoration: none;
            color: #051d40;
            font-size: 16px;

            &:hover {
                color: #9285f5;
            }
        }
    }
</style>
