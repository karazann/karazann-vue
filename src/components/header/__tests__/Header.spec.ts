import { shallowMount, RouterLinkStub } from '@vue/test-utils'
import Header from '../Header.vue'
import ProfileMenu from '../ProfileMenu.vue'

describe('Header', () => {
    const build = () => {
        const options = {
            data() {
                return {}
            },
            stubs: {
                NuxtLink: RouterLinkStub
            }
        }

        const wrapper = shallowMount(Header, options)

        return {
            wrapper,
            profileMenu: () => wrapper.find(ProfileMenu)
        }
    }

    it.skip('renders the component', () => {
        // arrange
        const { wrapper } = build()

        // assert
        expect(wrapper.html()).toMatchSnapshot()
    })

    it.skip('renders links properly', () => {
        // arrange
        const { wrapper } = build()
        wrapper.setProps({ signedIn: true })

        // assert
        expect(wrapper.html()).toMatchSnapshot()
    })

    it.skip('renders hidden state properly', () => {
        // arrange
        const { wrapper } = build()
        wrapper.setProps({ hidden: true })

        // assert
        expect(wrapper.isEmpty()).toBe(true)
    })
})
