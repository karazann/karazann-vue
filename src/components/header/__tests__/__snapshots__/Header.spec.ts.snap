// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders links properly 1`] = `
"<header class=\\"shadow\\">
  <div class=\\"container\\">
    <div class=\\"row inner\\">
      <v-brand></v-brand>
      <div class=\\"links\\">
        <div>
          <div class=\\"horizontal-list\\">
            <li>
              <n-link to=\\"/developers\\">Developers</n-link>
            </li>
            <li>
              <n-link to=\\"/blog\\">Blog</n-link>
            </li>
            <li>
              <n-link to=\\"/faq\\">FAQ</n-link>
            </li>
            <li>
              <n-link to=\\"/roadmap\\">Roadmap</n-link>
            </li>
            <li>
              <n-link to=\\"/auth/signin\\">Sign in</n-link>
            </li>
            <li>
              <n-link to=\\"/auth/signup\\" class=\\"btn\\">Sign up</n-link>
            </li>
          </div>
        </div>
        <!---->
      </div>
    </div>
  </div>
</header>"
`;

exports[`Header renders the component 1`] = `
"<header class=\\"shadow\\">
  <div class=\\"container\\">
    <div class=\\"row inner\\">
      <v-brand></v-brand>
      <div class=\\"links\\">
        <div>
          <div class=\\"horizontal-list\\">
            <li>
              <n-link to=\\"/developers\\">Developers</n-link>
            </li>
            <li>
              <n-link to=\\"/blog\\">Blog</n-link>
            </li>
            <li>
              <n-link to=\\"/faq\\">FAQ</n-link>
            </li>
            <li>
              <n-link to=\\"/roadmap\\">Roadmap</n-link>
            </li>
            <li>
              <n-link to=\\"/auth/signin\\">Sign in</n-link>
            </li>
            <li>
              <n-link to=\\"/auth/signup\\" class=\\"btn\\">Sign up</n-link>
            </li>
          </div>
        </div>
        <!---->
      </div>
    </div>
  </div>
</header>"
`;
